src/include/daos_types.h

```diff
@@ -675,11 +675,13 @@ typedef struct {
 	/*
 	 * Type of the value in an iod can be either a single type that is
 	 * always overwritten when updated, or it can be an array of EQUAL sized
-	 * records where the record is updated atomically. Note than an akey can
-	 * have both type of values, but to access both would require a separate
-	 * iod for each. If \a iod_type == DAOS_IOD_SINGLE, then iod_nr has to
-	 * be 1, and \a iod_size would be the size of the single atomic
-	 * value. The idx is ignored and the rx_nr is also required to be 1.
+	 * records where the record is updated atomically. Note that an akey can
+	 * only support one type of value which is set on the first update. If
+	 * user attempts to mix types in the same akey, the behavior is
```

Comment:
@mchaarawi can you please review the language here?  I modified it after discussion in F2F.  It is best not to promise (or even try to explain) any supported behavior (even if we support it).

Reply:
the language is fine, but i thought if we punch we would be able to "change" the type in future updates, no?

Reply:
Yes, the user will be able to do it after punch after the key has been deleted from the tree.   But the user has no control over that so it's simpler to just say the behavior is undefined rather than try to tell the user how to make the unsupported behavior work.

Reply:
Perhaps a better example is one where the user creates the value, then takes a snapshot, then punches the value.   It will never be able to change the type until the snapshot is removed even if it punches the object or key.

Reply:
yes i understand the issue. sounds good.